### PR TITLE
Add functions to get CIK from ticker or company name

### PR DESCRIPTION
**Pull request by daneads**
Adds a function to get a CIK from a ticker, and a function to get a list of CIKs and list of similar company names (if several company names match the string) for a given string

### CODE REVIEW

**Comment by jackmoody11** [comment]
I think this would be a great function to have! It might be useful to see if there would be a way to avoid using `BeautifulSoup` and get the CIK's directly from a GET request. Not sure if this is possible. Will look into possibilities.

**Comment by coyo8** [comment]
Yeah, I will also have look into. Basically, SEC site doesn't provide any API and so beautifulSoup is the only way unless we will have to rely on third-party services.

File: SECEdgar/crawler.py [hunk 1]
```diff
@@ -36,124 +37,109 @@ def save_in_directory(self, company_code, cik, priorto, doc_list,
...
             required_url = link_list_final[k].replace('-index.html', '')
-            txtdoc = required_url + ".txt"
-            docname = txtdoc.split("/")[-1]
+            txtdoc = required_url + '.txt'
+            docname = txtdoc.split('/')[-1]
             doc_list.append(txtdoc)
             doc_name_list.append(docname)
         return doc_list, doc_name_list
 
+    def get_ciks_for_company_name(self, company_name):
```
[the remark this answers is not in the document]
Comment (daneads) [reply SECEdgar/crawler.py]: @coyo8 I agree with the cached approach and am happy to revise this PR.

[the remark this answers is not in the document]
Comment (coyo8) [reply SECEdgar/crawler.py]: Thanks, @daneads :) Let me know if you need any help. We have updated the codebase quite a bit and you can have a look.

**Comment by jackmoody11** [comment]
I think it might be best to create this in a separate file (like cik.py). I used the code you wrote a while back and changed a few things. If you like any of it, feel free to use it for your PR.

```python
import requests
from bs4 import BeautifulSoup


def get_ciks_for_company_name(self, company_name):
    base = 'https://www.sec.gov/cgi-bin/cik_lookup'
    params = {'company': company_name}
    r = requests.post(base, params=params)
    data = r.text
    soup = BeautifulSoup(data, features='html.parser')
    cik_list = list(a.get_text() for a in soup.select('pre > a') if 'CIK=' in a['href'])
    company_list = list(str(a.next_sibling).strip()
                        for a in soup.select('pre > a') if 'CIK=' in a['href'])
    return cik_list, company_list

def get_cik_for_ticker(self, ticker):
    base = 'http://www.sec.gov/cgi-bin/browse-edgar'
    params = {'action': 'getcompany', 'CIK': ticker}
    r = requests.get(base, params=params)
    data = r.text
    soup = BeautifulSoup(data, features='html.parser')
    cik = str(soup.find('input', {'name': 'CIK'})['value'])
    return cik
```

**Comment by jackmoody11** [comment]
This was taken care of in #83 with the `CIKValidator` class (used by the CIK class).

**Pull request closed without merging**